crates/ruff_python_parser/src/string_token_flags.rs

```diff
@@ -0,0 +1,295 @@
+use std::fmt;
+
+use bitflags::bitflags;
+
+use ruff_text_size::{TextLen, TextSize};
+
+bitflags! {
+    /// Flags that can be queried to obtain information
+    /// regarding the prefixes and quotes used for a string literal
+    #[derive(Default, Debug, Copy, Clone, PartialEq, Eq, Hash)]
+    struct StringFlags: u8 {
+        /// The string uses double quotes (`"`).
+        /// If this flag is not set, the string uses single quotes (`'`).
+        const DOUBLE = 1 << 0;
+
+        /// The string is triple-quoted:
+        /// it begins and ends with three consecutive quote characters.
+        const TRIPLE_QUOTED = 1 << 1;
+
+        /// The string has a `u` or `U` prefix.
+        /// While this prefix is a no-op at runtime,
+        /// strings with this prefix can have no other prefixes set.
+        const U_PREFIX = 1 << 2;
+
+        /// The string has a `b` or `B` prefix.
+        /// This means that the string is a sequence of `int`s at runtime,
+        /// rather than a sequence of `str`s.
+        /// Strings with this flag can also be raw strings,
+        /// but can have no other prefixes.
+        const B_PREFIX = 1 << 3;
+
+        /// The string has a `f` or `F` prefix, meaning it is an f-string.
+        /// F-strings can also be raw strings,
+        /// but can have no other prefixes.
+        const F_PREFIX = 1 << 4;
+
+        /// The string has an `r` or `R` prefix, meaning it is a raw string.
+        /// F-strings and byte-strings can be raw,
+        /// as can strings with no other prefixes.
+        /// U-strings cannot be raw.
+        const R_PREFIX = 1 << 5;
```

Comment:
nit: we can have the same names as in `StringPrefix` for consistency

Reply:
I have a weak preference for keeping them "inconsistent", as I think it emphasises the fact that just because something has the `R_PREFIX` flag set doesn't necessarily mean that's the only prefix it has -- it might have other prefixes as well. In that way, it's semantically different to `StringPrefix` in an important way, since `StringPrefix` enumerates all the ways in which the prefixes can be validly _combined_, whereas the bitflag does not.